File: server/internal/dto/meta_mapping.go
package dto

// MetaMappingListResponse 元数据映射列表响应
type MetaMappingListResponse struct {
	TemplateKey  string            `json:"template_key"`
	TemplateName string            `json:"template_name"`
	Mappings     []MetaMappingItem `json:"mappings"`
}

// MetaMappingItem 元数据映射项
type MetaMappingItem struct {
	ID            uint   `json:"id"`
	SourceField   string `json:"source_field"`
	TargetField   string `json:"target_field"`
	FieldType     string `json:"field_type"`
	TransformRule string `json:"transform_rule"`
	IsActive      bool   `json:"is_active"`
	IsSystem      bool   `json:"is_system"`
	SortOrder     int    `json:"sort_order"`
}

type CreateMetaMappingRequest struct {
	TemplateKey   string `json:"template_key" binding:"required"`
	TemplateName  string `json:"template_name"` // 兼容旧前端传值，可忽略
	SourceField   string `json:"source_field" binding:"required"`
	TargetField   string `json:"target_field" binding:"required"`
	FieldType     string `json:"field_type" binding:"required"`
	TransformRule string `json:"transform_rule"`
	SortOrder     int    `json:"sort_order"`
}

type UpdateMetaMappingRequest struct {
	SourceField   string `json:"source_field"`
	TargetField   string `json:"target_field"`
	FieldType     string `json:"field_type"`
	TransformRule string `json:"transform_rule"`
	IsActive      *bool  `json:"is_active"`
	SortOrder     *int   `json:"sort_order"`
}

File: admin/src/types/metaMapping.ts
// MetaMapping 模块类型定义

/**
 * 映射模版信息
 */
export interface MetaMappingTemplate {
  id: number
  template_key: string
  template_name: string
  description: string
  mapping_count: number
}

/**
 * 创建模版请求
 */
export interface CreateMetaMappingTemplateRequest {
  template_key: string
  template_name: string
  description?: string
}

/**
 * 更新模版请求
 */
export interface UpdateMetaMappingTemplateRequest {
  template_name?: string
  description?: string
}

/**
 * 单个映射项
 */
export interface MetaMappingItem {
  id: number
  source_field: string
  target_field: string
  field_type: string
  transform_rule: string
  is_active: boolean
  is_system: boolean
  sort_order: number
}

/**
 * 映射列表响应（按模版分组）
 */
export interface MetaMappingListResponse {
  template_key: string
  template_name: string
  mappings: MetaMappingItem[]
}

/**
 * 创建映射请求
 */
export interface CreateMetaMappingRequest {
  template_key: string
  template_name: string
  source_field: string
  target_field: string
  field_type: string
  transform_rule?: string
  sort_order?: number
}

/**
 * 更新映射请求
 */
export interface UpdateMetaMappingRequest {
  source_field: string
  target_field: string
  field_type: string
  transform_rule?: string
  sort_order?: number
}
